esx_service/vmdk_ops.py

```diff
@@ -792,20 +822,34 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
         vol_name, datastore = parse_vol_name(full_vol_name)
     except ValidationError as ex:
         return err(str(ex))
-    if not datastore:
-        datastore = default_datastore
 
-    if not vmdk_utils.validate_datastore(datastore):
+    if datastore and not vmdk_utils.validate_datastore(datastore):
```

Comment:
Here "datastore==None" means the volume which user passed is a short name(no datastore is specified by user), so no need to validate "datastore" in this case.